test/Test/Toml/BiCode/Property.hs

```diff
@@ -0,0 +1,191 @@
+    <*> Toml.timeOfDay                       "timeOfDay"           .= btTimeOfDay
+    <*> Toml.arrayOf Toml._Int               "arrayOfInt"          .= btArray
+    <*> Toml.arraySetOf Toml._Word           "arraySetOfWord"      .= btArraySet
+    <*> Toml.arrayIntSet                     "arrayIntSet"         .= btArrayIntSet
+    <*> Toml.arrayHashSetOf Toml._Double     "arrayHashSetDouble"  .= btArrayHashSet
+    <*> Toml.arrayNonEmptyOf Toml._Text      "arrayNonEmptyOfText" .= btArrayNonEmpty
+    <*> Toml.nonEmpty (Toml.byteString "bs") "nonEmptyBS"          .= btNonEmpty
+    <*> Toml.list (Toml.bool "bool")         "listBool"            .= btList
+    <*> Toml.diwrap (Toml.zonedTime "nt.zonedTime")                .= btNewtype
+    <*> bigTypeSumCodec                                            .= btSum
+    <*> Toml.table bigTypeRecordCodec        "table-record"        .= btRecord
+
+_BigTypeSumA :: TomlBiMap BigTypeSum Integer
+_BigTypeSumA = Toml.prism BigTypeSumA $ \case
+    BigTypeSumA i -> Right i
+    other   -> Toml.wrongConstructor "BigTypeSumA" other
+
+_BigTypeSumB :: TomlBiMap BigTypeSum Text
+_BigTypeSumB = Toml.prism BigTypeSumB $ \case
+    BigTypeSumB n -> Right n
+    other    -> Toml.wrongConstructor "BigTypeSumB" other
+
+bigTypeSumCodec :: TomlCodec BigTypeSum
+bigTypeSumCodec =
+        Toml.match (_BigTypeSumA >>> Toml._Integer) "integer"
+    <|> Toml.match (_BigTypeSumB >>> Toml._Text)    "text"
+
+bigTypeRecordCodec :: TomlCodec BigTypeRecord
+bigTypeRecordCodec = BigTypeRecord
+    <$> Toml.arraySetOf Toml._Bool "rboolSet" .= btrBoolSet
+    <*> Toml.list bigTypeSumCodec "rnewtype" .= btrNewtypeList
+
+----------------------------------------------------------------------------
+-- Generator
+----------------------------------------------------------------------------
+
+genBigType :: Gen BigType
+genBigType = do
+    btBool          <- genBool
+    btInteger       <- genInteger
+    btNatural       <- fromInteger <$> genInteger
+    btInt           <- genInt
+    btWord          <- genWord
+    btDouble        <- genDouble
+    btFloat         <- genFloat
+    btText          <- genText
+    btString        <- genString
+    btBS            <- genBS
+    btLazyBS        <- L.fromStrict <$> genBS
+    btLocalTime     <- genLocal
+    btDay           <- genDay
+    btTimeOfDay     <- genHours
+    btArray         <- Gen.list (Range.constant 0 256) genInt
+    btArraySet      <- Gen.set (Range.constant 0 256) genWord
+    btArrayIntSet   <- genIntSet
+    btArrayHashSet  <- genHashSet genDouble
+    btArrayNonEmpty <- genNonEmpty genText
+    btNonEmpty      <- genNonEmpty genBS
+    btList          <- Gen.list (Range.constant 0 256) genBool
+    btNewtype       <- genNewType
+    btSum           <- genSum
+    btRecord        <- genRec
+    pure BigType {..}
+
+genInt :: Gen Int
+genInt = Gen.int (Range.constant 0 256)
+
+genWord :: Gen Word
+genWord = Gen.word (Range.constant 1 256)
```

Comment:
You can use `constantBounded` generator for `Int` and `Word` to avoid specifying boundaries manually. Unfortunately, won't fork for `Double` and `Float`, they don't have instances...

* https://hackage.haskell.org/package/hedgehog-0.6.1/docs/Hedgehog-Internal-Range.html#v:constantBounded

Reply:
@vrom911 Yeah, I think it's a good idea to move general-purpose generators (like `genWord`) into single module, because they are probably going to be reusable in several places (for example, I can see that `BiMap.Property` uses most of the same generators. It's a good idea to keep all basic generators in single place :slightly_smiling_face: